### PR TITLE
Fix reset of legacy 'Enable SpecFlowSingleFileGeneratorCustomTool' setting

### PR DESCRIPTION
**Pull request by rik-smeets**
This is done by separating registry registration of the single file generator and the VS extension option preference.

Registering the single file generator in the registry is still necessary I believe, see [Registering Single File Generators](https://docs.microsoft.com/en-us/visualstudio/extensibility/internals/registering-single-file-generators?view=vs-2019) over at Microsoft Docs. 

I also changed the default setting to `false`, which seems only logical for a legacy setting.

Resolves https://github.com/techtalk/SpecFlow/issues/1584.

I found it quite hard to test, but it seems to be working alright. Could you do an extensive test as well? Thanks!

### CODE REVIEW

**Comment by SabotageAndi** [comment]
Ok, you were faster than I thought this would take.

**Comment by rik-smeets** [comment]
@SabotageAndi Well, let's just hope everythings works as I intended before starting the party ;-)

**Comment by walliski** [comment]
@rik-smeets Is there any estimate of when this will be released?

**Comment by rik-smeets** [comment]
@walliski I think you should ask @SabotageAndi, I can't release a new version of the extension to the Marketplace.

**Comment by walliski** [comment]
Aah, yes. Didn't see who merged it, sorry about that. Lets see what SabotageAndi says. 👍

**Comment by SabotageAndi** [comment]
I will have a look at it in the next days.

**Comment by rik-smeets** [comment]
@SabotageAndi @walliski I finished testing the new version (and I'm using it at work too) and it's working fine! The legacy setting does not reset any longer after VS updates, and viewing the diff of a feature file does not dispose the GherkinLanguageService anymore on the opened feature file.

Sometimes when opening a solution, an exception occurs in the extension (it does continue working however). However, I checked with a colleague who does not have the new version installed and this exception occurred there as well. Might be due to a recent VS update, but is not related to my changes.

Long story short: in my opinion, a new version of the extension can be made available publicly :-).

**Comment by SabotageAndi** [comment]
Extension is release to the marketplace.

**Comment by LadaVarga** [comment]
Where we can change this settings to true: LegacyEnableSpecFlowSingleFileGeneratorCustomTool. Looks like 3 people cannot see specflow tests in their two different projects.

**Comment by david1995** [comment]
Hi @LadaVarga, you can enable `SpecFlowSingleFileGenerator` at [Tools] > [Options] > [SpecFlow] by setting the `Enable SpecFlowSingleFileGenerator Custom Tool` setting to true. Hope this helps.
